### PR TITLE
Feat/default leverage

### PR DESCRIPTION
**Pull request by tinsights**

### CODE REVIEW

File: src/modules/order.ts [hunk 1]
```diff
@@ -27,13 +27,31 @@ export class OrderModule extends BaseModule {
...
-      opts
-    );
+    if (params.setLeverage) {
+      return await wallet.sendTxs([
+        {
+          typeUrl: CarbonTx.Types.MsgSetLeverage,
+          value: {
+            creator: wallet.bech32Address,
+            market: params.market,
+            leverage: params.setLeverage.shiftedBy(18).toString(10),
```
Comment (tinsights) [inline src/modules/order.ts:37]: tbh haven't tested, but followed the format set in` leverage.ts`

File: src/modules/order.ts [hunk 2]
```diff
@@ -181,6 +199,8 @@ export namespace OrderModule {
     /** commission percents, input 10 for 10% */
     referralCommission?: number;
     referralKickback?: number;
+    /** default leverage to 3x for new trades */
```
Comment (GulfOfAlaska) [inline src/modules/order.ts:202]: maybe comment like:
leverage is 3x as default. this will help order creator set leverage for market to provided value.

**Review by GulfOfAlaska — APPROVED** [review]

File: src/modules/order.ts [hunk 3]
```diff
@@ -63,6 +82,18 @@ export class OrderModule extends BaseModule {
...
+      const leverage = params[0].setLeverage
+      msgs.unshift({
+        typeUrl: CarbonTx.Types.MsgSetLeverage,
+        value: {
+          creator: wallet.bech32Address,
+          market: params[0].market,
+          leverage: leverage.shiftedBy(18).toString(10),
+        },
+      })
+    }
```
Comment (tinsights) [inline src/modules/order.ts:95]: for TPSL orders. on the front end, setLeverage is set for the very first order, before the TP/SL order is added into the array. is this robust? initially checked every iteration in the `.map()` (previous commit).

will need to document behaviour on frontend

File: src/modules/order.ts (hunk 1 above)
Comment (stevenkhong) [reply src/modules/order.ts:39]: test before merging into SDK?

File: src/modules/order.ts (hunk 3 above)
Comment (stevenkhong) [reply src/modules/order.ts:95]: maybe we can have `params: (OrderModule.CreateOrderParams | OrderModule.SetLeverageParams) [], …` instead

File: src/modules/order.ts [hunk 4]
```diff
@@ -181,6 +212,11 @@ export namespace OrderModule {
     /** commission percents, input 10 for 10% */
     referralCommission?: number;
     referralKickback?: number;
+    /** leverage for new trades is defaulted to 3x.
```
Comment (stevenkhong) [inline src/modules/order.ts:215]: this context is unnecessary for the SDK imo, not relevant in a generic case as there is no 3x anywhere in the code. possibly lead to more confusion.

**Review by stevenkhong — CHANGES_REQUESTED** [review]

File: src/modules/order.ts (hunk 3 above)
Comment (tinsights) [reply src/modules/order.ts:95]: so for a single order we add leverage into the orderParams, but for TPSL / multi order we submit an array of orders + leading setLeverage ?

File: src/modules/order.ts (hunk 1 above)
Comment (tinsights) [reply src/modules/order.ts:39]: actually im not sure what i meant here lol. i tested locally linked sdk :p

File: src/modules/order.ts (hunk 3 above)
Comment (GulfOfAlaska) [reply src/modules/order.ts:95]: think should be fine

File: src/modules/order.ts (hunk 1 above)
Comment (GulfOfAlaska) [reply src/modules/order.ts:39]: ok gotta get the testers to try it

**Review by GulfOfAlaska — APPROVED** [review]

File: src/modules/order.ts (hunk 1 above)
Comment (stevenkhong) [reply src/modules/order.ts:39]: no i meant dev test locally with `yarn link`

Comment (tinsights) [reply src/modules/order.ts:39]: yes that's what i meant, i tested with `yarn link`ed sdk, to testnet, leverage got set.

**Pull request merged**